elk/utils/csv.py

```diff
@@ -0,0 +1,39 @@
+import csv
+from pathlib import Path
+from typing import Iterator, Callable, TextIO, TypeVar
+
+from datasets import DatasetDict
+
+from elk.evaluation.evaluate_log import EvalLog
+from elk.logging import save_debug_log
+from elk.training.train_log import ElicitLog
+
+"""A generic log type that contains a layer field
+The layer field is used to sort the logs by layer."""
+Log = TypeVar("Log", EvalLog, ElicitLog)
+
+
```

Comment:
this typevar has to be bounded on `EvalLog, ElicitLog` so you can sort by the layer properly with pyright